tests/core/e2e/test_s2ef.py

```diff
@@ -102,9 +110,14 @@ def smoke_test_train(
             # verify predictions from train and predict are identical
             energy_from_train = np.load(training_predictions_filename)["energy"]
             energy_from_checkpoint = np.load(predictions_filename)["energy"]
-            npt.assert_allclose(
-                energy_from_train, energy_from_checkpoint, rtol=1e-6, atol=1e-6
-            )
+            if not amp:
+                npt.assert_allclose(
+                    energy_from_train, energy_from_checkpoint, rtol=1e-6, atol=1e-6
```

Comment:
Are these amp and not amp precisions too low?

Reply:
What do you mean by too low? to high tolerance?

Reply:
Errors accumulate through the layers, so machine epsilon gets bumped a few orders. The constants that are used for non-amp do not work for amp. We could/should do a better analysis of if this is acceptable, but this is a whole side quest in of its own. 

I would be happy having at least some test for AMP (none existed before this), and work upwards from there?

Reply:
Ah thanks for the explanation! yeah no need to do that detailed of an analysis for this PR specifically.